irc/getters.go

```diff
@@ -373,6 +373,9 @@ func (client *Client) SetMode(mode modes.Mode, on bool) bool {
 func (client *Client) SetRealname(realname string) {
 	client.stateMutex.Lock()
 	client.realname = realname
+	if client.alwaysOn {
+		client.markDirty(IncludeRealname)
```

Comment:
Ah, there's a race here, `Client.markDirty()` is going to try to acquire a lock directly after this.